controllers/utils/utils.go

```diff
@@ -1,22 +1,66 @@
 package utils
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
 	"reflect"
 
 	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type IsvcDeploymentMode string
+
+var (
+	Serverless    IsvcDeploymentMode = "Serverless"
+	RawDeployment IsvcDeploymentMode = "RawDeployment"
+	ModelMesh     IsvcDeploymentMode = "ModelMesh"
 )
 
 const (
-	inferenceServiceDeploymentModeAnnotation      = "serving.kserve.io/deploymentMode"
-	inferenceServiceDeploymentModeAnnotationValue = "ModelMesh"
+	inferenceServiceDeploymentModeAnnotation = "serving.kserve.io/deploymentMode"
+	KserveConfigMapName                      = "inferenceservice-config"
 )
 
-func IsDeploymentModeForIsvcModelMesh(isvc *kservev1beta1.InferenceService) bool {
+func GetDeploymentModeForIsvc(ctx context.Context, cli client.Client, isvc *kservev1beta1.InferenceService) (IsvcDeploymentMode, error) {
+
+	// If ISVC specifically sets deployment mode using an annotation, return bool depending on value
 	value, exists := isvc.Annotations[inferenceServiceDeploymentModeAnnotation]
-	if exists && value == inferenceServiceDeploymentModeAnnotationValue {
-		return true
+	if exists {
+		switch value {
+		case string(ModelMesh):
+			return ModelMesh, nil
+		case string(Serverless):
+			return Serverless, nil
+		case string(RawDeployment):
+			return RawDeployment, nil
```

Comment:
Being this an uncontrolled annotation, I would recommend to add a default:
```suggestion
		case string(RawDeployment):
			return RawDeployment, nil
		default:
			return "", fmt.Errorf("the deployment mode '%s' of the Inference Service is invalid", value)
```